cmd/resolve-version/main_test.go

```diff
@@ -39,3 +40,41 @@ func TestParseObject(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.Errorf(t, err, "Failed to parse key")
 }
+
+func TestListS3Objects(t *testing.T) {
+	// Node
+	objects, err := listS3Objects("heroku-nodebin", "node")
```

Comment:
These tests depend on the `heroku-nodebin` bucket being available, and I'm not 100% happy about that but I'm not sure if mocking out all of the calls would actually end up testing anything here.

Reply:
i think that's fine, especially given how static the bucket is (i.e. changes are only additive). You might want to split the ones that require network into an integration test (ex. [the java-buildpack](https://github.com/heroku/java-buildpack/blob/master/jdk/integration_test.go#L1)). Then run with like:

```
$ go test ./... -tags=integration
```

Reply:
TIL about test tags